signer/src/network/libp2p/network.rs

```diff
@@ -505,4 +527,184 @@ mod tests {
+        context2
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key1));
+
+        // Add key2 to the known signers for signer1.
+        context1
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key2));
+
+        // Create the context for signer2.
+        let context3 = TestContext::builder()
+            .with_in_memory_storage()
+            .with_mocked_clients()
+            .modify_settings(|settings| {
+                settings.signer.private_key = key3;
+            })
+            .build();
+        // Add key1 and key2 to the known signers for signer 3. This simulates
+        // what an adversary signer might do, i.e. it would want to join with
+        // the known signers.
+        context3
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key1));
+        context3
+            .state()
+            .current_signer_set()
+            .add_signer(PublicKey::from_private_key(&key2));
+
+        let swarm1_addr = Multiaddr::random_memory();
+        let swarm2_addr = Multiaddr::random_memory();
+        let swarm3_addr = Multiaddr::random_memory();
+
+        // Create the two trusted swarms.
+        let mut swarm1 = SignerSwarmBuilder::new(&key1)
+            .enable_memory_transport(true)
+            .add_listen_endpoint(swarm1_addr.clone())
+            .add_seed_addrs(&[swarm2_addr.clone(), swarm3_addr.clone()])
+            .build()
+            .expect("Failed to build swarm 1");
+        let mut swarm2 = SignerSwarmBuilder::new(&key2)
+            .enable_memory_transport(true)
+            .add_listen_endpoint(swarm2_addr.clone())
+            .add_seed_addrs(&[swarm1_addr.clone(), swarm3_addr.clone()])
+            .build()
+            .expect("Failed to build swarm 2");
+        // Create the adversarial swarm.
+        let mut swarm3 = SignerSwarmBuilder::new(&key3)
+            .enable_memory_transport(true)
+            .add_listen_endpoint(swarm3_addr.clone())
+            .add_seed_addrs(&[swarm1_addr, swarm2_addr])
+            .build()
+            .expect("Failed to build swarm 3");
+
+        // Create the network liasons for the swarms (i.e. `MessageTransfer`
+        // instances).
+        let mut trusted1 = P2PNetwork::new(&context1);
+        let mut trusted2 = P2PNetwork::new(&context2);
+        let mut adversarial = P2PNetwork::new(&context3);
+
+        // Start the swarms.
+        let handle1 = tokio::spawn(async move {
+            swarm1.start(&context1).await.unwrap();
+        });
+        let handle2 = tokio::spawn(async move {
+            swarm2.start(&context2).await.unwrap();
+        });
+        let handle3 = tokio::spawn(async move {
+            swarm3.start(&context3).await.unwrap();
+        });
+
+        // The swarms are discovering themselves via mDNS, so we need to give
```

Comment:
nit: isn't `enable_mdns: false,` by default?

Reply:
yeah, and memory doesn't use mdns anyway, so [6cb92f5](https://github.com/stacks-network/sbtc/pull/1242/commits/6cb92f55a8a4c7bda25f8ee1e053083f1552b596)